codebasin/file_parser.py

```diff
@@ -126,15 +126,15 @@ def insert_directive_node(tree, line_group, logical_line):
         new_node.num_lines = line_group.line_count
         tree.insert(new_node)
 
-    def parse_file(self, *, summarize_only=True):
+    def parse_file(self, *, summarize_only=True, language=None):
```

Comment:
I had to look this up because I'd forgotten, but we're using `*` here to separate positional arguments from keyword arguments.  This is something that we do (or tried to do, at least) for every function that accepts keyword arguments.

With the `*` in place, a user can call `parse_file(summarize_only=True)` but cannot call `parse_file(True)`.  Without the `*`, both calls would work.  Using the `*` forces us to be more explicit about which keyword arguments we're setting at the call site, which improves readability.  It also ensures that we won't end up with function calls like `foo(True, False)` where it would be easy to mix up the order of the arguments.

Reply:
Ah okay, makes sense!